sdk/eventhubs/azure_messaging_eventhubs/src/common/mod.rs

```diff
@@ -78,63 +94,63 @@ impl ManagementInstance {
 
     pub async fn get_eventhub_partition_properties(
         &self,
-        eventhub: impl Into<String>,
-        partition_id: impl Into<String>,
+        eventhub: String,
+        partition_id: String,
     ) -> Result<EventHubPartitionProperties> {
-        let partition_id: String = partition_id.into();
-
         let mut application_properties: AmqpOrderedMap<String, AmqpValue> = AmqpOrderedMap::new();
-        application_properties.insert("name", eventhub.into());
-        application_properties.insert("partition", partition_id);
+        application_properties.insert(EVENTHUB_PROPERTY_NAME.to_string(), eventhub.into());
+        application_properties.insert(EVENTHUB_PROPERTY_PARTITION.to_string(), partition_id.into());
 
         let response = self
             .management
-            .call("com.microsoft:partition", application_properties)
+            .call(PARTITION_ENTITY_TYPE.to_string(), application_properties)
             .await?;
 
         // Look for the required response properties
-        if !response.contains_key("type")
-            || !response.contains_key("last_enqueued_sequence_number_epoch")
+        if !response.contains_key(EVENTHUB_PARTITION_PROPERTIES_TYPE.to_string())
```

Comment:
Why does `contains_key` need an owned `String` as opposed to a borrowed `&str`? Same with `get()` below. These seems like the wrong API choice, or do we not own the callee (hard to tell from this PR)?

Reply:
Can I do `impl Into<&K>`? Or would this be `impl AsRef<Into<K>>`?

Reply:
This is an interesting question. 

Currently `contains_key` is defined as `contains_key(key: K)`, I can change it to `contains__key(key:&K)` but that still won't allow you to pass in a string slice.

Instead this forces you to write:

`&AmqpSymbol::from("key")`

Reply:
Oh - last night in my sleep, I found what I believe is the perfect answer. 

```rust
   pub fn contains_key(&self, key: impl PartialEq<K>) -> bool {
```

The negative is that it uses an `impl Trait` for the parameter, the positive is that it works for ANY type which implements PartialEq, which means it works with `String`, `&str`, `AmqpSymbol`, whatever.